lib/improver/weighted_blend.py

```diff
@@ -0,0 +1,111 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing Weighted Blend classes."""
+import warnings
+
+import numpy as np
+import iris
+
+
+class BasicWeightedAverage(object):
+    """Apply a Basic Weighted Average to a cube."""
+
+    def __init__(self, coord, coord_adjust=None):
+        """Set up for a Basic Weighted Average Blending plugin
+
+        Args:
+            coord : string
+                     The name/s of a coordinate dimension/s in the cube
+            coord_adjust : Function to apply to the coordinate after
+                           collapsing the cube to correct the values
+                           for example for time windowing and
+                           cycle averaging the follow function would
+                           adjust the time coordinates
+            e.g. coord_adjust = lambda pnts: pnts[len(pnts)/2]
+        """
+        self.coord = coord
+        self.coord_adjust = coord_adjust
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        return (
+            '<BasicWeightedAverage: coord = {0:s}>').format(self.coord)
+
+    def process(self, cube, weights=None):
+        """Calculated weighted mean across the chosen coord
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+
+        Returns:
+            result : iris.cube.Cube
+
+        """
+        if not isinstance(cube, iris.cube.Cube):
+            raise ValueError('The first argument must be an instance of ' +
+                             'iris.cube.Cube but is' +
+                             ' {0:s}'.format(type(cube)))
+        if not cube.coords(self.coord):
+            raise ValueError('The coord for this plugin must be ' +
+                             'an existing coordinate in the input cube')
+        # Find the coords dimension.
+        # If coord is a scalar_coord try adding it
+        collapse_dim = cube.coord_dims(self.coord)
+        if not collapse_dim:
+            msg = ('Could not find collapse dimension, ' +
+                   'will try adding it')
+            warnings.warn(msg)
+            cube = iris.util.new_axis(cube, self.coord)
+            collapse_dim = cube.coord_dims(self.coord)
+        # supply weights as an array of weights whose shape matches the cube
+        weights_array = None
+        if weights is not None:
+            if np.array(weights).shape != cube.coord(self.coord).points.shape:
+                raise ValueError('The weights array must match the shape ' +
```

Comment:
It would be worth including the values of `np.array(weights).shape` and `cube.coord(self.coord).points.shape` within the error message.